src/Cardano/CoinSelection/Random.hs

```diff
@@ -49,66 +49,168 @@ import Data.Functor
---    algorithm for this step.)
---
--- 2. The algorithm first makes a random selection for each output from the
---    UTxO, processing the biggest output first and proceeding in a descending
---    order.  If the selection is not successful largest-first fallback kicks
---    in.  If the selection is successful for each output then the improvement
---    is tried for each selection, once again starting from the selection made
---    for the biggest output. The improvement is tried for the next biggest
---    output's selection. An output is considered an improvement when:
---
---    (a)  It doesn’t exceed a specified upper limit.
---    (b)  Adding the new output gets us closer to the ideal change value.
---    (c)  It doesn’t exceed a maximum number of transaction inputs.
---
--- This algorithm follows three principles:
---
--- @
--- **Self organisation principle 1**
--- Random selection has a high probability of picking dust outputs precisely
--- when there is a lot of dust in the UTxO.
--- @
---
--- @
--- **Self organisation principle 2**
--- If for each payment request for value `x` we create a change output roughly
--- of the same value `x`, then we will end up with a lot of change outputs in
--- our UTxO of size `x` precisely when we have a lot of payment requests of
--- size `x`
--- @
---
--- @
--- **Self organisation principle 3**
--- Searching the UTxO for additional entries to improve our change output is
--- only useful if the UTxO contains entries that are sufficiently small enough.
--- But precisely when the UTxO contains many small entries, it is less likely
--- that a randomly chosen UTxO entry will push the total above the upper bound
--- we set.
--- @
+-- | An implementation of the __Random-Improve__ coin selection algorithm.
+--
+-- = Overview
+--
+-- The __Random-Improve__ coin selection algorithm works in __two phases__, by
+-- /first/ selecting UTxO entries /at random/ to pay for each of the given
+-- outputs, and /then/ attempting to /improve/ upon each of the selections.
+--
+-- === Phase 1: Random Selection
+--
+-- __In this phase, the algorithm randomly selects a minimal set of UTxO__
+-- __entries to pay for each of the given outputs.__
+--
+-- During this phase, the algorithm:
+--
+--   *  processes outputs in /descending order of coin value/.
+--
+--   *  maintains a /remaining UTxO set/, initially equal to the given
+--      /UTxO set/ parameter.
+--
+-- For each output of value __/v/__, the algorithm /randomly/ selects entries
+-- from the /remaining UTxO set/, until the total value of selected entries is
+-- greater than or equal to __/v/__. The selected entries are then associated
+-- with that output, and removed from the /remaining UTxO set/.
+--
+-- This phase ends when every output has been associated with a selection of
+-- UTxO entries.
+--
+-- However, if the remaining UTxO set is completely exhausted before all
+-- outputs can be processed, the algorithm terminates and delegates to the
```

Comment:
maybe it is better to say instead of delegates `falls back to` (just hunch;-)

Reply:
Fair enough!

Fixed in 60258d5fbc5bc272b41d8f68df358a5ad75c47da.